### PR TITLE
Bind Dqlite to a local-cloud address if possible at bootstrap

### PR DESCRIPTION
**Pull request by manadart**
JUJU-4611

To-date we have bound Dqlite to the loopback address (127.0.0.1) at bootstrap, and rebound it to a local-cloud address when entering HA.

What has been observed, particularly in testing by solutions QA, is that the Dqlite rebind step and the attendant churn is causing a period where the controller is inaccessible.

In this change, we will bind Dqlite to a local-cloud address at bootstrap, if we can determine that the address is unique in that scope. This means there is no disruption at all when entering HA, and HA is established much sooner.

If there is no unique local-cloud address determined, we simply fall back to the prior behaviour of using the loopback IP, and rebinding later based on the peer-grouper's address determination.

## QA steps

- Bootstrap to LXD (the default profile/bridge should ensure a unique address).
- `juju ssh -m controller 0`.
- `sudo cat /var/lib/juju/dqlite/info.yaml` will show we're not bound to the loopback.
- `juju enable-ha`.
- `juju debug-log -m controller` will indicate entry into HA _without_ disconnecting as it did before.

## Documentation changes

None.

## Bug reference

https://bugs.launchpad.net/juju/+bug/2015371

### CODE REVIEW

File: database/node.go [hunk 1]
```diff
@@ -209,6 +209,49 @@ func (m *NodeManager) WithTracingOption() app.Option {
...
+
+		sysAddrs, err := nic.Addresses()
+		if err != nil || len(sysAddrs) == 0 {
+			continue
+		}
+
+		for _, addr := range sysAddrs {
+			addrs = append(addrs, corenetwork.NewMachineAddress(addr.IP().String()))
+		}
+	}
```
Comment (SimonRichardson) [inline database/node.go:244]: Nit: You could exit out early if there are no addrs to make it easier to read the code.

**Review by SimonRichardson — COMMENTED** [review]
Code looks fine. Will Q&A now.

**Review by SimonRichardson — APPROVED** [review]
Q&A was good.

File: database/node.go (hunk 1 above)
Comment (manadart) [reply database/node.go:244]: True, but I prefer to have the log line and fall-back option in a single place.

**Pull request merged**